documentation/server-reflection-tutorial.md

```diff
@@ -0,0 +1,166 @@
+# gRPC Server Reflection Tutorial
+
+gRPC Server Reflection provides information about publicly-accessible gRPC
+services on a server, and assists clients at runtime with constructing RPC
+requests and responses without precompiled service information. It is used by
+the gRPC command line tool (gRPC CLI), which can be used to introspect server
+protos and send/receive test RPCs. Reflection is only supported for
+proto-based services.
+
+## Enable Server Reflection
+
+gRPC-Java Server Reflection is implemented in
+[ProtoReflectionService.java](https://github.com/grpc/grpc-java/tree/master/services/src/main/java/io/grpc/protobuf/service/ProtoReflectionService.java)
```

Comment:
Oh, I had intended the links would be dropped as well. Linking to the project isn't helpful, and I'd rather not link to the implementation ProtoReflectionService. If anything, we could link to the JavaDoc.